backend/app/api/endpoints/adapter/subscription_follows.py

```
@@ -0,0 +1,351 @@
+
+
+discover_router = APIRouter()
+
+
+@discover_router.get("/discover", response_model=DiscoverSubscriptionsListResponse)
+def discover_subscriptions(
+    sort_by: str = Query("popularity", description="Sort by: 'popularity' or 'recent'"),
+    search: Optional[str] = Query(None, description="Search query"),
+    page: int = Query(1, ge=1, description="Page number"),
+    limit: int = Query(50, ge=1, le=100, description="Items per page"),
+    db: Session = Depends(get_db),
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)